### DISCUSSION TITLE
Getting VLAN tag from zeek.

### DISCUSSION THREAD

**Discussion by OzzyKampha** [Security-Onion-Solutions/securityonion#8632] [Unsupported Versions] [answered]
Is there a way to get VLAN tags from zeek? i saw that zeek have a script to add vlan to con.logs, but have someone tested it?

**Comment by dougburks** [comment, marked as answer]
From https://docs.securityonion.net/en/2.3/bpf.html#vlan:
> Warning
Please note that [Zeek](https://docs.securityonion.net/en/2.3/zeek.html#zeek) and [Stenographer](https://docs.securityonion.net/en/2.3/stenographer.html#stenographer) should correctly analyze traffic on a VLAN but won’t log the actual VLAN tags due to the way that [AF-PACKET](https://docs.securityonion.net/en/2.3/af-packet.html#af-packet) works:
https://github.com/J-Gras/zeek-af_packet-plugin/issues/9
https://github.com/google/stenographer/issues/211

**Reply by OzzyKampha** [reply]
If i change to suricata for metadata, does it parse vlan currectly?

**Reply by dougburks** [reply]
To be clear, Zeek parses VLAN-tagged traffic correctly, it just doesn't log the actual VLAN tags. 

Suricata parses VLAN-tagged traffic and logs the actual VLAN tags. You can find the VLAN tag in the `message` field.

**Reply by OzzyKampha** [reply]
Thank you for the reply. Is it necessary to have Suricata as the "metadata mode" for the vlans to get logged or does it also do it as the ids?

**Reply by dougburks** [reply]
If you are running Zeek for metadata and Suricata for NIDS, then the Suricata NIDS alerts will have the VLAN tag but no other logs will.
